client/src/models/Person.js

```diff
@@ -30,21 +44,41 @@ export default class Person extends Model {
 
 	static nameDelimiter = ','
 
-	static schema = {
-		name: '',
-		get status() { return Person.STATUS.ACTIVE },
-		country: '',
-		rank: '',
-		gender: 'MALE',
-		phoneNumber: '',
-		endOfTourDate: null,
-		biography: '',
-		get role() { return Person.ROLE.PRINCIPAL },
-		position: {},
-		...Model.schema,
-	}
-
-	static autocompleteQuery = "uuid, name, role, rank, position { uuid, name, code, organization { uuid, shortName }, location {uuid, name} }"
+	static yupSchema = yup.object().shape({
+		name: yup.string().nullable().default(''),
+		// not actually in the database, but used for validation
+		firstName: yup.string().nullable()
+			.when('role', (role, schema) => (
+				Person.isAdvisor({role}) ? schema.required(`You must provide the ${fieldLabels.firstName}`) : schema.nullable()
+			)).default('').label(fieldLabels.firstName),
+		// not actually in the database, but used for validation
+		lastName: yup.string().nullable().uppercase().required(`You must provide the ${fieldLabels.lastName}`).default('').label(fieldLabels.lastName),
+		domainUsername: yup.string().nullable().default('').label(fieldLabels.domainUsername),
+		emailAddress: yup.string().nullable().email()
+			.when('role', (role, schema) => (
+				Person.isAdvisor({role}) ? schema.required(`You must provide the ${fieldLabels.emailAddress}`) : schema.nullable()
+			)).default('').label(fieldLabels.emailAddress),
+		country: yup.string().nullable().required(`You must provide the ${fieldLabels.country}`).default('').label(fieldLabels.country),
```

Comment:
Can/should we make use of yup to enforce that certain attributes (country, rank, gender, role, status) are assigned permitted values (i.e country is part of Settings.fields.advisor.person.countries if advisor, etc...)?